test/agent.js

```diff
@@ -1176,11 +1178,25 @@ function assertMetadata (t, payload) {
   t.equal(payload.service.name, 'some-service-name')
   t.deepEqual(payload.service.runtime, { name: 'node', version: process.versions.node })
   t.deepEqual(payload.service.agent, { name: 'nodejs', version: agentVersion })
-  t.deepEqual(payload.system, {
-    hostname: os.hostname(),
-    architecture: process.arch,
-    platform: process.platform
-  })
+
+  const expectedSystemKeys = ['hostname', 'architecture', 'platform']
+  if (inContainer) expectedSystemKeys.push('container')
+
+  t.deepEqual(Object.keys(payload.system), expectedSystemKeys)
+  t.equal(payload.system.hostname, os.hostname())
+  t.equal(payload.system.architecture, process.arch)
+  t.equal(payload.system.platform, process.platform)
+
+  if (inContainer) {
+    t.deepEqual(Object.keys(payload.system.container), ['id'])
+    t.equal(typeof payload.system.container.id, 'string')
+    t.ok(/^[\da-f]{64}$/.test(payload.system.container.id))
+  } else {
+    // dummy asserts to ensure that the t.plan is as expected
```

Comment:
Maybe just change the `asserts` count on this function when in a container?

Reply:
Doh, you're right. I was thinking it could change between tests, but obviously it can't

Reply:
Done: #998